src/objc-renderer.ts

```diff
@@ -163,35 +163,58 @@ function toKeywordString(keyword:ObjC.Keyword):string {
   return keyword.name + Maybe.match(toKeywordArgumentString, emptyString, keyword.argument);
 }
 
+
+function toOptionalPreprocessorOpeningCodeString(method:ObjC.Method):string {
+  if(method.preprocessors.length == 0) {
+    return '';
+  }
+  return method.preprocessors.map(function(object) {return object.openingCode}).join('\n') + '\n';
+}
+
+function toOptionalPreprocessorClosingCodeString(method:ObjC.Method):string {
+  if(method.preprocessors.length == 0) {
+    return '';
+  }
+  return  '\n' + method.preprocessors.map(function(object) {return object.closingCode}).join('\n');
+}
+
 function toClassMethodHeaderString(method:ObjC.Method):string {
   const methodComments = method.comments.map(toCommentString).join('\n');
   const methodCommentsSection = codeSectionForCodeStringWithoutExtraSpace(methodComments);
   const compilerAttributesString = method.compilerAttributes.length > 0 ? " " + method.compilerAttributes.join(" ") : "";
 
-  return methodCommentsSection + '+ (' + toTypeString(method.returnType) + ')' + method.keywords.map(toKeywordString).join(' ') + compilerAttributesString + ';';
+  return toOptionalPreprocessorOpeningCodeString(method) +
```

Comment:
Instead of repeating this four times, maybe we add a `statementWrappedInPreprocessorCondition` helper?